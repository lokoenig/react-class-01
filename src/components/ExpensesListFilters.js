import React from "react";
import { connect } from "react-redux";
import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";


import { setFilterText, sortByDate, sortByAmount } from "../actions/filters";

const ExpensesListFilters = (props)=>(
<>
    <form>
        <input 
        type="text" 
        value= {props.filters.text} 
        onChange={ (e)=>{
            props.dispatch(setFilterText(e.target.value))
        }} />
        <select
        value= {props.filters.sortBy}
        onChange={ (e)=>{
            if ('date' === e.target.value){
                props.dispatch(sortByDate());
            }else{
                props.dispatch(sortByAmount());
            }
        }}
        >
            <option value="date" >Date</option>
            <option value="amount" >Amount</option>
        </select>
    </form>
</>
);

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

export default connect(mapStateToProps)(ExpensesListFilters);