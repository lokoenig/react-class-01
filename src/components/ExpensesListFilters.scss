// override the 100% width of default stylesheet
.react-datepicker-wrapper .react-datepicker__input-container {
    width: fit-content;
    padding-right: 2rem;
}