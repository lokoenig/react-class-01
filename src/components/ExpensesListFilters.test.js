import React from "react";
import { render, screen, waitFor, within } from '@testing-library/react';
import user from '@testing-library/user-event';

import {ExpensesListFilters} from "./ExpensesListFilters";
import { FilterTestData , nullFilterTestData} from "../fixtures/filter";

let setFilterText, sortByDate, sortByAmount, setFilterDateRange;
// let getByText, asFragment;

beforeEach( ()=>{
    setFilterText = jest.fn();
    sortByDate = jest.fn();
    sortByAmount = jest.fn();
    setFilterDateRange = jest.fn();
    render(<ExpensesListFilters
        filters={FilterTestData} 
        setFilterText={setFilterText}
        sortByDate={sortByDate}
        sortByAmount={sortByAmount}
        setFilterDateRange={setFilterDateRange}
        />);

});

test('renders ExpensesList Filters component with normal data', () => {
    const { asFragment, getByText } = render(<ExpensesListFilters
        filters={FilterTestData}
        setFilterText={setFilterText}
        sortByDate={sortByDate}
        sortByAmount={sortByAmount}
        setFilterDateRange={setFilterDateRange}
    />);
  
    expect(asFragment()).toMatchSnapshot();
});

test('renders ExpensesList Filters component with null data', () => {
    const { asFragment, getByText } = render(<ExpensesListFilters
        filters={nullFilterTestData}
        setFilterText={setFilterText}
        sortByDate={sortByDate}
        sortByAmount={sortByAmount}
        setFilterDateRange={setFilterDateRange}
    />);

    expect(asFragment()).toMatchSnapshot();
});