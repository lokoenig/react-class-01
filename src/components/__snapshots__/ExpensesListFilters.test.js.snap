// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ExpensesList Filters component with normal data 1`] = `
<DocumentFragment>
  <form>
    <input
      type="text"
      value=""
    />
    <select>
      <option
        value="date"
      >
        Date
      </option>
      <option
        value="amount"
      >
        Amount
      </option>
    </select>
    <p>
      start: 12/31/1979
    </p>
    <p>
      end: 12/31/2028
    </p>
    <label
      for="double-pick"
    >
      Date Range
    </label>
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <span
          aria-live="polite"
          class="react-datepicker__aria-live"
          role="alert"
        />
        <input
          class=""
          id="double-pick"
          placeholder="Select a display range"
          type="text"
          value="12/31/1979 - 12/31/2028"
        />
        <button
          aria-label="Close"
          class="react-datepicker__close-icon"
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`renders ExpensesList Filters component with null data 1`] = `
<DocumentFragment>
  <form>
    <input
      type="text"
      value=""
    />
    <select>
      <option
        value="date"
      >
        Date
      </option>
      <option
        value="amount"
      >
        Amount
      </option>
    </select>
    <p>
      start: 12/17/2019
    </p>
    <p>
      end: 02/28/2023
    </p>
    <label
      for="double-pick"
    >
      Date Range
    </label>
    <div
      class="react-datepicker-wrapper"
    >
      <div
        class="react-datepicker__input-container"
      >
        <span
          aria-live="polite"
          class="react-datepicker__aria-live"
          role="alert"
        />
        <input
          class=""
          id="double-pick"
          placeholder="Select a display range"
          type="text"
          value="12/17/2019 - 02/28/2023"
        />
        <button
          aria-label="Close"
          class="react-datepicker__close-icon"
          tabindex="-1"
          type="button"
        />
      </div>
    </div>
  </form>
</DocumentFragment>
`;
